Doc/library/math.rst

```diff
@@ -284,6 +286,17 @@ Number-theoretic and representation functions
    :class:`~numbers.Integral` (usually an integer). Delegates to
    :meth:`x.__trunc__() <object.__trunc__>`.
 
+.. function:: ulp(x)
+
+   Return the value of the least significant bit of a float *x*, such that the
+   first float bigger than *x* is ``x + ulp(x)``.  Then, given an expected
```

Comment:
This description doesn't work for negative `x`: e.g., if `x = -1.0`. Maybe the documentation should describe what happens only for positive `x`, and then comment that for negative `x`, the result is just `ulp(-x)` (or `ulp(abs(x))`)?

Reply:
We should also explicitly state the behaviour for zeros, infinities and nans. `ulp` is not a standard function, and even the experts don't agree on how it should be defined. See for example [this article](http://ljk.imag.fr/membres/Carine.Lucas/TPScilab/JMMuller/ulp-toms.pdf).

Reply:
done